src/training/vqa_fine_tune.py

```diff
@@ -0,0 +1,316 @@
+from training.scheduler import cosine_lr
+from training.train import AverageMeter
+import evaluate
+from sklearn import preprocessing
+import numpy as np
+
+from datasets import load_dataset_builder
+from datasets import load_dataset
+
+class VQATextDataset(Dataset):
+    def __init__(self, df, split, transforms, labelencoder, tokenizer=None):
+        self.df = df
+        self.transforms = transforms
+        self.tokenize = tokenizer
+        self.labels = labelencoder.transform(df['multiple_choice_answer'])
+    def __len__(self):
+        return len(self.df)
+
+    def __getitem__(self, idx):
+        item = self.df.iloc[idx]
+        img_path = item["image"]["path"]
+        image = Image.open(str(img_path))
+        text = item["question"]
+        label = self.labels[idx]
+        return {
+            'image': self.transforms(image),
+            'text': self.tokenize([text])[0],
+            'label': torch.tensor(label)
+        }
+
+def get_task_dataloaders(path, transforms, labelencoder, args):
+    tokenizer = get_tokenizer(args.model)
+    dataloaders = {}
+    
+    for split in ["train", "validation"]:
+        dataset_train = load_dataset(path, split=split, cache_dir = "./sample_data")
+        dataset_df = dataset_train.to_pandas()
+
+        class_id = []
+        questions = []
+        images = []
+        answers = []
+        for index, row in dataset_df.iterrows():
+          if(row['multiple_choice_answer'] in answer_set):
+            class_id.append(row['question_id'])
+            questions.append(row['question'])
+            images.append(row['image'])
+            answers.append(row['multiple_choice_answer'])
+        class_id = np.array(class_id)
+        questions = np.array(questions)
+        images = np.array(images)
+        answers = np.array(answers)
+
+        dataset_df = pd.DataFrame({'question_id': class_id, 'question': questions, 'image': images, 'multiple_choice_answer': answers})
+        #dataset_df = dataset_df[0:12800]
+        b_size = args.batch_size
+        if(split == "validation"):
+            b_size = args.batch_size * 20
+            dataset_df = dataset_df[0:12800]
+            dataset = VQATextDataset(dataset_df,
+                split,
+                transforms,
+                labelencoder,              
+                tokenizer=tokenizer,
+            )
+        dataloader = DataLoader(
+            dataset,
+            batch_size=b_size,
+            shuffle=True,
+            num_workers=args.workers,
+            pin_memory=True,
+            drop_last=True,
+        )
+        dataloaders[split] = dataloader
+
+    return dataloaders
+
+class CLIPMultimodalClassifier(nn.Module):
+    def __init__(self, encoder, embed_dim, num_labels):
+        super().__init__()
+
+        self.encoder = encoder
+        
+        self.fc1 = nn.Linear(embed_dim * 2, 1536) #size of answer space
+        self.lnorm = nn.LayerNorm(1536)
+        self.fc2 = nn.Linear(1536, num_classes)
```

Comment:
Taking `num_labels` in but assigning `num_classes`?

Reply:
whoops! That's a weird typo hmm...